src/test/java/streams/part2/exercise/Exercise1.java

```diff
@@ -0,0 +1,158 @@
+package streams.part2.exercise;
+
+import lambda.data.Employee;
+import lambda.data.JobHistoryEntry;
+import lambda.data.Person;
+import org.junit.Test;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.*;
+import static org.junit.Assert.assertEquals;
+
+@SuppressWarnings({"ConstantConditions", "unused"})
+public class Exercise1 {
+
+    @Test
+    public void calcTotalYearsSpentInEpam() {
+        List<Employee> employees = getEmployees();
+
+        Long hours = employees
+                .stream()
+                .flatMap(employee -> employee.getJobHistory().stream())
+                .filter(jobHistoryEntry -> "EPAM".equals(jobHistoryEntry.getEmployer()))
+                .map(JobHistoryEntry::getDuration)
+                .map(Integer::longValue)
+                .reduce((aLong, aLong2) -> aLong + aLong2)
+                .orElseThrow(RuntimeException::new);
+
+        assertEquals(19, hours.longValue());
+    }
+
+    @Test
+    public void findPersonsWithQaExperience() {
+        List<Employee> employees = getEmployees();
+
+        Set<Person> workedAsQa = employees
+                .stream()
+                .filter(employee -> employee
+                        .getJobHistory()
+                        .stream()
+                        .map(JobHistoryEntry::getPosition)
+                        .collect(toSet())
+                        .contains("QA"))
```

Comment:
This is very heavy operation (collecting to set for simple "contains").
It's better to use stream's  `.anyMatch(...)` after mapping to `position`.